java-checks/src/main/java/org/sonar/java/checks/ReleaseSensorsCheck.java

```diff
@@ -55,17 +60,34 @@ enum AcquireReleaseSensor {
       this.acquireMethodMatcher = MethodMatchers.create().ofTypes(sensorClass).names(acquireMethod).withAnyParameters().build();
       this.releaseMethodMatcher = MethodMatchers.create().ofTypes(sensorClass).names(releaseMethod).withAnyParameters().build();
     }
+
+    AcquireReleaseSensor(String acquireSensorClass, String acquireMethod, String releaseSensorClass, String releaseMethod) {
+      this.acquireMethodMatcher = MethodMatchers.create().ofTypes(acquireSensorClass).names(acquireMethod).withAnyParameters().build();
+      this.releaseMethodMatcher = MethodMatchers.create().ofTypes(releaseSensorClass).names(releaseMethod).withAnyParameters().build();
+    }
   }
 
   private static class AcquireReleaseStatus {
     List<Tree> acquireInvocations = new LinkedList<>();
     boolean released = false;
   }
 
+  public static final String ACQUIRE = "acquire";
   public static final String RELEASE = "release";
   private AcquireReleaseStatus[] statuses;
 
-  public ReleaseSensorsCheck() {
+  @Override
+  public void setContext(JavaFileScannerContext context) {
+    super.setContext(context);
+    initStatuses();
+  }
+
+  @Override
+  public void leaveFile(JavaFileScannerContext context) {
+    Arrays.stream(statuses)
+      .filter(status -> !status.released)
+      .forEach(status -> status.acquireInvocations.forEach(mit -> reportIssue(mit, "Make sure to release this sensor after use.")));
+
     initStatuses();
```

Comment:
Since we do the init work in the `setContext` now, we can simply set the field to `null` here - otherwise, we will initialize the field twice in most cases.